tdrs-frontend/src/components/FileUpload/FileUpload.jsx

```diff
@@ -89,6 +93,22 @@ function FileUpload({ section, setLocalAlertState }) {
     filereader.onloadend = (evt) => {
       /* istanbul ignore next */
       if (!evt.target.error) {
+        // Validate file extension before proceeding
+        const re = /(\.txt|\.ms\d{2}|\.ts\d{2,3})$/i
```

Comment:
personally, i wouldn't mind having a list of `ALLOWED_EXTENSIONS` somewhere, which could be used in a function instead of the regex and also `.join()`ed in the error message. The regex is less code but less readable/reusable imo

Reply:
I like the idea of having a list of extensions as const in comparison to regex here

Reply:
How would we do that when the `.ts` and `.ms` extensions can have two or three digits appended to them with duplicate digits allowed? That would be a list with ~1200 elements. We could generate that list each time instead of making it a static const, but that is kind of expensive. Or is there some fact about those extensions I am missing? Something like: "the first number is always a zero." Some fact that would reduce the combination space?